src/librustc/middle/check_const.rs

```diff
@@ -474,9 +474,13 @@ impl<'a, 'tcx, 'v> Visitor<'v> for CheckCrateVisitor<'a, 'tcx> {
                                     self.tcx, ex, ExprTypeChecked) {
                                 Ok(_) => {}
                                 Err(msg) => {
-                                    span_err!(self.tcx.sess, msg.span, E0020,
+                                    let ret = span_err!(self.tcx.sess, msg.span, E0020,
                                               "{} in a constant expression",
-                                              msg.description())
+                                              msg.description());
+                                    span_help!(self.tcx.sess, msg.span,
+                                               "Dividing by 0 always leads to an error and is \
+                                                certainly not what you want");
```

Comment:
In this case, I found it useful because there is no case you might want to divide by 0. So instead of removing it, do you have a better sentence maybe ?